windows/windows_update_install/prepare_msu_file.yml

```diff
@@ -41,26 +60,56 @@
 - name: "Copy the .msu file to local disk of guest OS"
   include_tasks: ../utils/win_execute_cmd.yml
   vars:
+    win_execute_cmd_no_log: "{{ hide_secrets | default(false) }}"
     win_powershell_cmd: >-
       {{ win_nfs_mount_cmd }};
-      Copy-Item -Path {{ msu_file_src_path }} -Include *.msu -Destination {{ msu_dir_path }}  -ErrorAction Stop;
-      net use {{ drive_letter_new }}: /delete
-    win_execute_cmd_no_log: "{{ hide_secrets | default(false) }}"
+      $file = Get-ChildItem -Path {{ msu_file_nfs_path }} | where Name -like '{{ msu_file_name_reg }}';
+      if ($file) {Copy-Item -Path $file.FullName -Destination {{ msu_dest_dir }} -ErrorAction Stop;};
+      net use {{ drive_letter_new }}: /delete;
+      Write-Host $file.FullName
 
-- name: "Get the .msu file name"
-  include_tasks: ../utils/win_execute_cmd.yml
-  vars:
-    win_powershell_cmd: >- 
-      Get-ChildItem -Path {{ msu_dir_path }} -Include *.msu -Name -ErrorAction Stop;
+- name: "Check if the .msu file exists on source"
+  when:
+    - win_powershell_cmd_output.stdout_lines is defined
+    - win_powershell_cmd_output.stdout_lines | length > 0
+  block:
+    - name: "Set facts if the .msu file exists on source"
+      ansible.builtin.set_fact:
+        msu_file_exists_on_src: true
```

Comment:
You can directly find out the msu file with a select filter, instead of looping all items. For example,
```
{{ win_powershell_cmd_output.stdout_lines | select('search', '(?i)windows.*\\.msu$') }}
```
This will return a list. If no msu file found, it will be an empty list.

Reply:
Updated. Thanks.